app/src/features/datasets/components/DatasetArtifacts.tsx

```diff
@@ -145,7 +158,7 @@ const DatasetAttachments: React.FC<IDatasetAttachmentsProps> = (props) => {
             color="primary"
             startIcon={<Icon path={mdiLockPlus} size={1} />}
             onClick={() => setOpenApplySecurity(true)}
-            disabled={selectedArtifacts.length > 0 ? false : true}>
+            disabled={hasAdministrativePermissions && selectedArtifacts.length > 0 ? false : true}>
```

Comment:
Nit: Can be simplified to `disabled={!hasAdministrativePermissions || selectedArtifacts.length === 0}`